[15.0][MIG]purchase_unreconciled

AaronHForgeFlow opened the pull request:
@ForgeFlow

AaronHForgeFlow committed:
[ADD] purchase_unreconciled

AaronHForgeFlow committed:
[IMP] purchase_unreconciled: black, isort, prettier

AaronHForgeFlow committed:
[15.0][MIG]purchase_unreconciled

AaronHForgeFlow committed:
[FIX] purchase_unreconciled: do not create empty write-offs
[FIX] purchase_unreconciled: assign purchase order to the write-off created

rafaelbn commented:
/ocabot migration purchase_unreconciled

AaronHForgeFlow committed:
[15.0][FIX] purchase_unreconciled: reconcile exit condition

AaronHForgeFlow committed:
[IMP] purchase_unreconciled: allow companies to decide automatic reconcile or not

AaronHForgeFlow commented:
![forever_alone_pr](https://user-images.githubusercontent.com/19620251/178739901-b7e9aafb-ebb2-471f-b5bb-7e81b936fc1b.jpg)

AaronHForgeFlow committed:
[FIX]purchase_unreconciled: reconcile by currency

JuanyDForgeflow reviewed (approved):
Functional review LGTM

AaronHForgeFlow committed:
[FIX] purchase_unreconciled: if journals have no residual but are not full reconcile create a full reconcile

AaronHForgeFlow committed:
[IMP] purchase_unreconciled: Add tolerance parameter

AaronHForgeFlow committed:
[FIX] purchase_unreconciled: multicompany in base search unreconciled domain

AaronHForgeFlow committed:
[FIX]purchase_unreconciled: company context when seeing unreconciled journal items

AaronHForgeFlow committed:
[FIX] purchase_unreconciled: account domain

AaronHForgeFlow committed:
[IMP] purchase_unreconciled: reconcile by groups

AaronHForgeFlow committed:
[FIX]purchase_unreconciled: safety check in write-off

AaronHForgeFlow committed:
[FIX] purchase_reconcile: write-off currency was missing

AaronHForgeFlow committed:
[IMP] purchase_unreconciled: add hook for custom write-off date

AaronHForgeFlow committed:
[FIX] purchase_unreconcile: base unreconciled domain mathcing standard

AaronHForgeFlow committed:
pur unrec

AaronHForgeFlow committed:
[IMP] purchase_unreconciled: amount unreconciled in tree view

AaronHForgeFlow committed:
[FIX] purchase_unreconciled: do the write-off on the latest date of the journals involved

[IMP] purchase_unreconciled: choose residual field

[FIX] purchase_unreconciled: unreconciled exception wizard

AaronHForgeFlow committed:
[FIX] purchase_unreconciled: do not show warning unreonciled message when reconcile criteria is not met
